rocket-okapi/src/gen.rs

```diff
@@ -33,15 +33,21 @@ impl OpenApiGenerator {
             // TODO do this outside add_operation
             op.operation.operation_id = Some(op_id.trim_start_matches(':').replace("::", "_"));
         }
-        match self.operations.entry((op.path, op.method)) {
-            HashEntry::Occupied(e) => {
-                let (path, method) = e.key();
-                panic!(
-                    "An OpenAPI operation has already been added for {} {}",
-                    method, path
-                );
+        match self.operations.entry(op.path.clone()) {
+            MapEntry::Occupied(mut e) => {
+                let map = e.get_mut();
+                if map.insert(op.method, op.operation).is_some() {
+                    // This will trow a warning if 2 routes have the same path and method
+                    // This is allowed by Rocket when a ranking is given for example: `#[get("/user", rank = 2)]`
+                    // See: https://rocket.rs/v0.4/guide/requests/#forwarding
+                    println!("Warning: Operation replaced for {}:{}", op.method, op.path);
```

Comment:
I think we can get rid of the `clone()` here:
```suggestion
        match self.operations.entry(op.path) {
            MapEntry::Occupied(mut e) => {
                let map = e.get_mut();
                if map.insert(op.method, op.operation).is_some() {
                    // This will trow a warning if 2 routes have the same path and method
                    // This is allowed by Rocket when a ranking is given for example: `#[get("/user", rank = 2)]`
                    // See: https://rocket.rs/v0.4/guide/requests/#forwarding
                    println!("Warning: Operation replaced for {}:{}", op.method, e.key());
```

Reply:
That is indeed nicer